neural_modelling/src/neuron/implementations/neuron_impl_standard.h

```diff
@@ -227,69 +235,82 @@ static bool neuron_impl_do_timestep_update(index_t neuron_index,
 
-    for (int i = 0; i < NUM_EXCITATORY_RECEPTORS; i++) {
-        total_exc += exc_input_values[i];
-    }
-    for (int i = 0; i < NUM_INHIBITORY_RECEPTORS; i++) {
-        total_inh += inh_input_values[i];
-    }
+        // Call functions to obtain exc_input and inh_input
+        input_t* exc_input_values = input_type_get_input_value(
+                exc_value, input_type, NUM_EXCITATORY_RECEPTORS);
+        input_t* inh_input_values = input_type_get_input_value(
+                inh_value, input_type, NUM_INHIBITORY_RECEPTORS);
 
-    // Call functions to get the input values to be recorded
-    neuron_recording_record_accum(GSYN_EXC_RECORDING_INDEX, neuron_index, total_exc);
-    neuron_recording_record_accum(GSYN_INH_RECORDING_INDEX, neuron_index, total_inh);
+        // Sum g_syn contributions from all receptors for recording
+        REAL total_exc = 0;
+        REAL total_inh = 0;
 
-    // Call functions to convert exc_input and inh_input to current
-    input_type_convert_excitatory_input_to_current(
-            exc_input_values, input_type, voltage);
-    input_type_convert_inhibitory_input_to_current(
-            inh_input_values, input_type, voltage);
+        for (int i = 0; i < NUM_EXCITATORY_RECEPTORS; i++) {
+            total_exc += exc_input_values[i];
+        }
+        for (int i = 0; i < NUM_INHIBITORY_RECEPTORS; i++) {
+            total_inh += inh_input_values[i];
+        }
+
+        // Do recording if on the first step
+        if (i == n_steps_per_timestep) {
+            neuron_recording_record_accum(V_RECORDING_INDEX, neuron_index, voltage);
+            neuron_recording_record_accum(GSYN_EXC_RECORDING_INDEX, neuron_index, total_exc);
+            neuron_recording_record_accum(GSYN_INH_RECORDING_INDEX, neuron_index, total_inh);
+        }
 
-    external_bias += additional_input_get_input_value_as_current(
-            additional_input, voltage);
+        // Call functions to convert exc_input and inh_input to current
+        input_type_convert_excitatory_input_to_current(
+                exc_input_values, input_type, voltage);
+        input_type_convert_inhibitory_input_to_current(
+                inh_input_values, input_type, voltage);
 
-    // update neuron parameters
-    state_t result = neuron_model_state_update(
-            NUM_EXCITATORY_RECEPTORS, exc_input_values,
-            NUM_INHIBITORY_RECEPTORS, inh_input_values,
-            external_bias, neuron);
+        external_bias += additional_input_get_input_value_as_current(
+                additional_input, voltage);
 
-    // determine if a spike should occur
-    bool spike = threshold_type_is_above_threshold(result, threshold_type);
+        // update neuron parameters
+        state_t result = neuron_model_state_update(
+                NUM_EXCITATORY_RECEPTORS, exc_input_values,
+                NUM_INHIBITORY_RECEPTORS, inh_input_values,
+                external_bias, neuron);
```

Comment:
@pabogdan has used this successfully.  The trick here is that the model doesn't need to know, because the model is told that it is running at a smaller timestep outside, and the code here handles that it is being called multiple times in a single timestep.

Reply:
Ah, okay - I wasn't completely sure that was what was going on, but now I see it.  

I wonder whether it's a good idea to document somewhere relating to the new parameter that there can be quantisation problems with particular neuron models, and warn the user somehow to be careful when using this feature?

Reply:
It would be good to document it, but probably not in the code, as it is pretty hidden from the user just now really...

Reply:
OK, that's fine - just as long as we don't forget to document it somewhere.